sysconfig/cloudinit.go

```diff
@@ -263,6 +263,68 @@ func cloudDatasourcesInUse(configFile string) (*cloudDatasourcesInUseResult, err
 	return res, nil
 }
 
+// cloudDatasourcesInUseForDir considers all files in a directory as individual
+// cloud-init config files, and analyzes all datasources in use for each file
+// and returns their union. It does not distinguish between mentioned,
+// explicitly allowed, or explicitly disallowed, but it does follow cloud-init's
+// logic for determining the overwriting of properties such that if a lexcially
+// earlier file sets datasource_list: [], then a later file sets
+// datasource_list: [foo], then foo is used instead and the explicit disallowing
+// is ignored/overwritten, but if no other file sets datasource_list, then it is
+// treated as if the config explicitly disallows no datasources.
```

Comment:
Can we make it more explicit (assuming that my understanding is correct):

```suggestion
// logic for determining the overwriting of properties. So, for example, if a file
// sets datasource_list: [] and no other file processed later (files are processed
// in lexical order) sets this property to another value, it will be treated as if
// the config explicitly disallows no datasources. If, on the other hand, a file
// processed later sets datasource_list: [foo], then foo is used instead and the
// explicit disallowing is ignored/overwritten
```

Reply:
yes that is more clear, thanks for the suggestion